### PR TITLE
refactor: use more transactions

### PR DESCRIPTION
**Pull request by koivunej**
this will make sure that all db access is transactional, which it
already was. hooray! also, I can add new stuff, which only works in
transactions.

### CODE REVIEW

File: crates/pathfinder/src/storage/schema/fixtures/mod.rs [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
-            )
-            .unwrap();
+            rusqlite::named_params! {
+                ":number": block.number.0,
+                ":hash": block.hash.0.as_be_bytes(),
+                ":root": block.root.0.as_be_bytes(),
+                ":timestamp": block.timestamp.0,
+            },
+        )
+        .unwrap();
```
Comment (koivunej) [inline crates/pathfinder/src/storage/schema/fixtures/mod.rs:24]: This seems like an unintended de-identation due to shorter variable.

File: crates/pathfinder/src/storage/state.rs [hunk 2]
```diff
@@ -1274,38 +1289,26 @@ mod tests {
...
-                    }
+                    with_default_blocks(|tx, blocks| {
+                        for block in blocks {
+                            let result = StarknetBlocksTable::get(tx, block.number.into())
+                                .unwrap()
+                                .unwrap();
+
+                            assert_eq!(result, block);
+                        }
+                    })
```
Comment (koivunej) [inline crates/pathfinder/src/storage/state.rs:1300]: These were the only ones I found the energy to do, but probably something similar is needed elsewhere.

**Review by Mirko-von-Leipzig — APPROVED** [review]
Nice! Thanks for persevering!

**Review by CHr15F0x — APPROVED** [review]

File: crates/pathfinder/src/storage/state.rs [hunk 3]
```diff
@@ -1509,48 +1463,36 @@ mod tests {
...
-                    StarknetBlocksTable::insert(&connection, block).unwrap();
-                }
-
-                StarknetBlocksTable::reorg(&connection, StarknetBlockNumber::GENESIS).unwrap();
-
-                assert_eq!(
-                    StarknetBlocksTable::get(&connection, StarknetBlocksBlockId::Latest).unwrap(),
-                    None
-                );
+                    assert_eq!(
```
Comment (kkovaacs) [inline crates/pathfinder/src/storage/state.rs:1469]: Aren't we missing `StarknetBlocksTable::reorg(...)` here?

Comment (koivunej) [reply crates/pathfinder/src/storage/state.rs:1469]: Excellent find.

Comment (kkovaacs) [reply crates/pathfinder/src/storage/state.rs:1469]: Ah, it's there...

**Review by kkovaacs — APPROVED** [review]

**Pull request merged**